src/niscope/system_tests/test_system_niscope.py

```diff
@@ -31,9 +31,9 @@ def test_read(session):
     test_num_channels = 2
     session.configure_vertical(test_voltage, niscope.VerticalCoupling.AC)
     session.configure_horizontal_timing(50000000, test_record_length, 50.0, 1, True)
-    wfm, wfm_infos = session.channels[test_channels].read(test_record_length)
-    assert len(wfm) == test_num_channels * test_record_length
-    assert len(wfm_infos) == test_num_channels
+    waveform, waveforms = session.channels[test_channels].read(test_record_length)
```

Comment:
Do we need to do Fancy Read to match Fancy Fetch (eventually after Fancy Fetch is done)?

Reply:
I think we do. Though once Fancy Fetch is done, I think the same template will be able to be reused with very minor changes.

I think in general, once Fancy Fetch for Scope is done and Fancy Fetch Multiple is done for DCPower, we should sit down, possibly with driver API experts and see if there are any other low effort improvements we can make to the APIs.